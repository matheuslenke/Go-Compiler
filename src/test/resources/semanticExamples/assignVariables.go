package samples

import fmt "fmt"



func testeAtribuicao() {
	var e int = -1 // +
	var numero int = 300
	numero = e

	var floatVar float32 = 5.0
	var floatVar2 float32 = 6.0
	floatVar = floatVar2
	floatVar2 = floatVar

}
