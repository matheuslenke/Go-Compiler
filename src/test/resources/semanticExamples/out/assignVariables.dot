digraph {
graph [ordering="out"];
node0[label="source_file"];
node1[label="import_spec"];
node2[label="Println@1"];
node3[label="parameters"];
node4[label="(string) arg@2"];
node3 -> node4;
node2 -> node3;
node5[label="result"];
node2 -> node5;
node1 -> node2;
node6[label="Scanln@3"];
node7[label="parameters"];
node8[label="(string) arg@4"];
node7 -> node8;
node6 -> node7;
node9[label="result"];
node6 -> node9;
node1 -> node6;
node0 -> node1;
node10[label="(func) testeAtribuicao@5"];
node11[label="parameters"];
node10 -> node11;
node12[label="block"];
node13[label="var_list"];
node14[label="="];
node15[label="(int) e@6"];
node14 -> node15;
node16[label="(int) int@1"];
node14 -> node16;
node13 -> node14;
node12 -> node13;
node17[label="var_list"];
node18[label="="];
node19[label="(int) numero@7"];
node18 -> node19;
node20[label="(int) int@300"];
node18 -> node20;
node17 -> node18;
node12 -> node17;
node21[label="assign_list_node"];
node22[label="="];
node23[label="(int) numero@7"];
node22 -> node23;
node24[label="(int) e@6"];
node22 -> node24;
node21 -> node22;
node12 -> node21;
node25[label="var_list"];
node26[label="="];
node27[label="(float) floatVar@8"];
node26 -> node27;
node28[label="(float) real@5.0"];
node26 -> node28;
node25 -> node26;
node12 -> node25;
node29[label="var_list"];
node30[label="="];
node31[label="(float) floatVar2@9"];
node30 -> node31;
node32[label="(float) real@6.0"];
node30 -> node32;
node29 -> node30;
node12 -> node29;
node33[label="assign_list_node"];
node34[label="="];
node35[label="(float) floatVar@8"];
node34 -> node35;
node36[label="(float) floatVar2@9"];
node34 -> node36;
node33 -> node34;
node12 -> node33;
node37[label="assign_list_node"];
node38[label="="];
node39[label="(float) floatVar2@9"];
node38 -> node39;
node40[label="(float) floatVar@8"];
node38 -> node40;
node37 -> node38;
node12 -> node37;
node10 -> node12;
node0 -> node10;
}
