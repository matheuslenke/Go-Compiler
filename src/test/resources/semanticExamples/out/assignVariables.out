[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:




Variables table:
Entry 0 -- name: fmt, line: 3, type: no_type, scopeName: global
Entry 1 -- name: Println, line: 3, type: no_type, scopeName: global
Entry 2 -- name: arg, line: 3, type: string, scopeName: Println
Entry 3 -- name: Scanln, line: 3, type: no_type, scopeName: global
Entry 4 -- name: arg, line: 3, type: string, scopeName: Scanln
Entry 5 -- name: testeAtribuicao, line: 7, type: func, scopeName: global
Entry 6 -- name: e, line: 8, type: int, scopeName: testeAtribuicao
Entry 7 -- name: numero, line: 9, type: int, scopeName: testeAtribuicao
Entry 8 -- name: floatVar, line: 12, type: float, scopeName: testeAtribuicao
Entry 9 -- name: floatVar2, line: 13, type: float, scopeName: testeAtribuicao




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.745 s
[INFO] Finished at: 2022-02-24T20:53:42-03:00
[INFO] ------------------------------------------------------------------------
